cloud/blockstore/libs/storage/service/volume_session_actor_mount.cpp

```diff
@@ -920,8 +944,11 @@ TVolumeSessionActor::TMountRequestProcResult TVolumeSessionActor::ProcessMountRe
         return {{}, false};
     }
 
-    if (mountSeqNumber != clientInfo->MountSeqNumber) {
-        // If mountSeqNumber has changed in MountVolumeRequest from client
+    if (mountSeqNumber != clientInfo->MountSeqNumber ||
+        fillSeqNumber != clientInfo->FillSeqNumber ||
+        fillGeneration != clientInfo->FillGeneration) {
+        // If mountSeqNumber, fillSeqNumber or fillGeneration
+        // has changed in MountVolumeRequest from client
```

Comment:
Need to check fillSeqNumber and fillGeneration here. Otherwise, consider the case when the client that is already rw-mounted comes again with other fillSeqNumber/fillGeneration. In this case the code that validates these parameters will not be executed and we might accept client with wrong fillGeneration or outdated fillSeqNumber.

Relocation works correctly without this check because different iterations of disk replication have different client ids. But i don't think we can rely on this.